src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size + 1 <= elements.length) {
+            elements[size] = value;
+        } else {
+            Object[] moreElements = new Object[(int) (elements.length * 1.5)];
+            System.arraycopy(elements, 0, moreElements, 0, elements.length);
+            moreElements[size] = value;
+            elements = moreElements;
+        }
```

Comment:
[Resize the array in a separate method.](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#resize-the-array-in-a-separate-method)